### PR TITLE
0.18 FEM

### PR DESCRIPTION
**Pull request by berndhahnebach**
- [x] Branch rebased on latest master `git pull --rebase upstream master`
- [x] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)

played with FEM on the weekend and found some more things to fix for 0.18 They are not critical, but they are bugs.

bernd

### CODE REVIEW

File: src/Mod/Fem/femresult/resulttools.py [hunk 1]
```diff
@@ -40,6 +41,12 @@ def purge_results(analysis):
                 analysis.Document.removeObject(m.Mesh.Name)
             analysis.Document.removeObject(m.Name)
     FreeCAD.ActiveDocument.recompute()
+    # if analysis typ check is used result mesh without result obj is created in the analysis
```
Comment (luzpaz) [inline src/Mod/Fem/femresult/resulttools.py]: s/typ/type/

File: src/Mod/Fem/femresult/resulttools.py [hunk 2]
```diff
@@ -40,6 +41,12 @@ def purge_results(analysis):
...
+    # we could run into trouble in one loop because we gone delete objects and try to access them later
```
Comment (luzpaz) [inline src/Mod/Fem/femresult/resulttools.py]: s/gone/will/

File: src/Mod/Fem/femtest/testresult.py [hunk 3]
```diff
@@ -222,6 +222,36 @@ def test_read_frd_massflow_networkpressure(self):
...
+        # fcc_print(round(mises, 4))
+        self.assertEqual(round(mises, 4), expected_mises, "Calculated von Mises stress is not the expected value.")
+
+    def test_stress_prinzipal(self):
+        expected_prinzipal = (-178.0076, -194.0749, -468.9075, 145.4499)
+        from feminout.importToolsFem import calculate_principal_stress as pr
+        prin = pr(self.get_stress_values())
+        rounded_prin = (round(prin[0], 4), round(prin[1], 4), round(prin[2], 4), round(prin[3], 4))
+        # fcc_print(rounded_prin)
+        self.assertEqual(rounded_prin, expected_prinzipal, "Calculated prinzipal stresses are not the expected values.")
```
Comment (luzpaz) [inline src/Mod/Fem/femtest/testresult.py]: s/prinzipal/principal/ ?

Comment (berndhahnebach) [reply src/Mod/Fem/femtest/testresult.py]: done

File: src/Mod/Fem/femresult/resulttools.py (hunk 1 above)
Comment (berndhahnebach) [reply src/Mod/Fem/femresult/resulttools.py]: done

**Comment by berndhahnebach** [comment]
fixed unit test failure

**Comment by wwmayer** [comment]
Merged.

**Pull request closed without merging**